### DISCUSSION TITLE
Has KEDA seen any performance issues from relying on HPA?

### DISCUSSION THREAD

**Discussion by ellistarn** [kedacore/keda#1005] [General]
I'm trying to understand the highly fragmented autoscaling space. I think it's awesome that KEDA implemented using custom metrics and the HPA, but it struck me that Knative could do the same. I reached out to them, and they gave me a laundry lists of why not to use the HPA: https://github.com/knative/serving/issues/9087

Have you seen similar issues with the HPA? Is there an opportunity to unify some of these projects and upstream any performance improvements?

**Comment by zroubalik** [comment]
Hi, for KEDA releated scaling HPA we don't see any perf issue. Knative autoscaler needed more fine graned configuration, because it is scaling on requests, so it makes sense to implement custom scaler for Knative.
